capstone-frontend/src/components/SearchBookModal.jsx

```diff
@@ -0,0 +1,238 @@
+import React, { Component } from 'react';
+import { Button, Form, Spinner, Modal, Col, Row } from 'react-bootstrap'
+
+class SearchBookModal extends Component {
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      fetchingBooks: false, // For Spinner
+      showModal: false,
+      typingTimeout: 0,
+      searchTerm: "",
+      searchResults: [],
+      displayBooks: false,
+      addedBooksIDs: [],
+      addedBooks: [],
+    }
+  }
+
+  getBooks = async (searchTerm) => {
+
+    this.setState({ fetchingBooks: true })
+
+    let searchResults;
+
+    if (searchTerm === "") {
+      searchResults = [];
+
+      this.setState({ searchResults, displayBooks: false, fetchingBooks: false })
+    }
+    else {
+      searchResults = await fetch(`/api/search?searchTerm=${searchTerm}&maxResults=${4}`)
+        .then(response => response.json())
+        .catch(err => alert(err));
+
+      if (typeof searchResults === "undefined") {
+        searchResults = [];
+      }
+
+      this.setState({ searchResults, displayBooks: true, fetchingBooks: false })
+    }
+  }
+
+  handleSearchTermChange = (event) => {
+
+    if (this.state.typingTimeout) {
+      clearTimeout(this.state.typingTimeout);
+    }
+
+    this.setState({
+      searchTerm: event.target.value,
+      typingTimeout: setTimeout(async () => {
+        await this.getBooks(this.state.searchTerm)
+      }, 500)
+    })
+  }
+
+  addBookToList = (book) => {
+
+    if (this.props.type === "club") {
```

Comment:
What does the addBookToList() do in the case of a club? Clubs don't have booklists, just the one current book they're reading.

Reply:
This code basically makes it into a singleton array for the Club object.